File: frontend/src/types/appointment.ts
export enum RecurrenceInterval {
  Daily = 0,
  Weekly = 1,
  Monthly = 2
}

export interface Appointment {
  id: number;
  title: string;
  description: string;
  startTime: string;
  endTime: string;
  location: string;
  attendees: string;
  type: string;
  userId: number;
  
  // Recurrence properties
  isRecurring: boolean;
  recurrenceInterval: RecurrenceInterval;
  recurrenceEndDate?: string;
  parentAppointmentId?: number;
}

export interface NewAppointment {
  title: string;
  description: string;
  startTime: string;
  endTime: string;
  location: string;
  attendees: string;
  type: string;
  
  // Recurrence properties
  isRecurring: boolean;
  recurrenceInterval: RecurrenceInterval;
  recurrenceEndDate?: string;
}
// Define appointment types and their colors
export const APPOINTMENT_TYPES = [
  { id: 'meeting', label: 'Meeting', color: '#4285F4' },       // Blue
  { id: 'personal', label: 'Personal', color: '#0F9D58' },     // Green
  { id: 'reminder', label: 'Reminder', color: '#F4B400' },     // Yellow
  { id: 'deadline', label: 'Deadline', color: '#DB4437' },     // Red
  { id: 'travel', label: 'Travel', color: '#9C27B0' },         // Purple
  { id: 'health', label: 'Health', color: '#00BCD4' },         // Cyan
  { id: 'social', label: 'Social', color: '#FF9800' },         // Orange
  { id: 'other', label: 'Other', color: '#757575' }            // Gray
];

// Helper function to get color for appointment type
export const getAppointmentColor = (type?: string): string => {
  if (!type) return '#757575'; // Default gray
  const appointmentType = APPOINTMENT_TYPES.find(t => t.id === type);
  return appointmentType?.color || '#757575';
};

File: backend/DisprzTraining/Models/Appointment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace DisprzTraining.Models
{
    public class Appointment
    {
        public int Id { get; set; }

        [Required]
        public string Title { get; set; }

        public string Description { get; set; }

        public string Location { get; set; }

        public string Attendees { get; set; }

        [Required]
        public DateTime StartTime { get; set; }

        [Required]
        public DateTime EndTime { get; set; }
        [Required]
        public string Type { get; set; }

        // Foreign key
        public int UserId { get; set; }

        // Navigation property - should NOT have [Required] attribute
        [JsonIgnore] // Add this to prevent serialization issues
        public virtual User? User { get; set; }
        // New properties for recurrence
        public bool IsRecurring { get; set; } = false;
        
        public RecurrenceInterval RecurrenceInterval { get; set; } = RecurrenceInterval.Daily;
        
        public DateTime? RecurrenceEndDate { get; set; }
        
        // For tracking instances of a recurring series
        public int? ParentAppointmentId { get; set; }
    }
    
    public enum RecurrenceInterval
    {
        Daily = 0,
        Weekly = 1,
        Monthly = 2
    }
}